### ISSUE TITLE
Could you help remove the high severity vulnerability introduced by package node-forge?

### ISSUE THREAD

**Issue by paimon0715** [libp2p/js-peer-id#153]
Hi, @vasco-santos @jacobheun, there is a high severity vulnerability introduced in your package **peer-id**:
### Issue Description
A vulnerability [**CVE-2020-7720**](https://snyk.io/vuln/SNYK-JS-NODEFORGE-598677) detected in package **node-forge<0.10.0** is transitively referenced by **_peer-id@0.12.5_**. We noticed that such a vulnerability has been removed since **_peer-id@0.14.3_**.
  
 However, peer-id's popular previous version **_peer-id@0.12.5_** ([5,680 downloads per week](https://www.npmjs.com/package/peer-id/v/0.12.5?activeTab=versions)) is still transitively referenced by a large amount of latest versions of active and popular downstream projects (about **225 downstream projects**, e.g., @aragon/cli 7.1.6, snet-sdk-web 2.0.0-beta.0, @gny/cli 1.0.90, @augurproject/tools 2.1.13, @charged-particles/protocol-subgraph 1.2.7, @51nodes/decentralized-schema-registry@0.1.1, etc.).
As such, issue [**CVE-2020-7720**](https://snyk.io/vuln/SNYK-JS-NODEFORGE-598677) can be propagated into these downstream projects and expose security threats to them.
   
   These projects cannot easily upgrade **peer-id** from version **_0.12.5_** to _**(>=0.14.3)**_. For instance, **_peer-id@0.12.5_** is introduced into the above projects via the following package dependency paths:
(1)`@51nodes/decentralized-schema-registry@0.1.1 ➔ @evan.network/api-blockchain-core@2.20.0 ➔ @evan.network/dbcp@1.11.2 ➔ ipfs-api@26.1.2 ➔ peer-id@0.12.5 ➔ libp2p-crypto@0.16.3 ➔ node-forge@0.9.2`
    **......**
    
The projects such as **ipfs-api**, which introduced peer-id@0.12.5, are not maintained anymore. These unmaintained packages can neither upgrade **peer-id** nor be easily migrated by the large amount of affected downstream projects.  
 On behalf the downstream users, could you help us remove the vulnerability from package peer-id@0.12.5?

### Suggested Solution

Since these inactive projects set a version constaint **0.12.\*** for **peer-id** on the above vulnerable dependency paths, if **peer-id** removes the vulnerability from 0.12.5 and releases a new patched version **_peer-id@0.12.6_**, such a vulnerability patch can be automatically propagated into the 225 affected downstream projects.

In **_peer-id@0.12.6_**, you can kindly try to perform the following upgrade:
`libp2p-crypto ~0.16.1 ➔ ~0.19.0`;  
**Note:**
**_libp2p-crypto@0.19.0(>=0.19.0)_** directly depends on node-forge@0.10.0 (a vulnerability [**CVE-2020-7720**](https://snyk.io/vuln/SNYK-JS-NODEFORGE-598677) patched version)

Thank you for your help.

Best regards,
Paimon

**Comment by jacobheun** [comment]
Hi @paimon0715, typically we don't maintain versions this far back but the weekly download rate is rather high. I think the right place to fix this is in libp2p-crypto, as users may be installing it via other libp2p libraries. I'll investigate feasibility of an upgrade patch there.

**Comment by jacobheun** [comment]
This has been patched in libp2p-crypto@0.16.4 so any new installs of peer-id 0.12.x will come with node-forge 0.10.

```
peer-id@0.12.5
└─┬ libp2p-crypto@0.16.4
  └── node-forge@0.10.0 
```
